analyzer/src/test/java/org/jboss/gm/analyzer/alignment/io/SettingsFileIOTest.java

```diff
@@ -39,20 +39,22 @@ public void testSCMNameWithNoSCM() throws IOException {
 
     @Test
     public void testSCMNameWithSCMHigher() throws IOException {
+        File testPath = new File(TestUtils.class.getClassLoader().getResource("").getPath()).getParentFile().getParentFile()
+                .getParentFile();
+        File testSettings = new File(testPath, "tmp" + File.separator + "tests");
+        testSettings.mkdirs();
+        File settingsGradle = new File(testSettings, "settings.gradle");
+        settingsGradle.createNewFile();
+        SettingsFileIO.writeProjectNameIfNeeded(testSettings);
+        // Clean up so we can rerun and still succeed.
+        FileUtils.deleteQuietly(testSettings);
+    }
+
+    @Test(expected = ManipulationUncheckedException.class)
+    public void testFindGitDir() throws IOException {
         File tmpFolder = folder.newFolder();
         File subdir = new File(tmpFolder, "subfolder");
-        File root = new File(TestUtils.class.getClassLoader().getResource("").getPath())
-                .getParentFile().getParentFile().getParentFile().getParentFile();
-        // Weird hack to work around debugging in IntelliJ versus running in Gradle project build.
-        if (root.getName().equals("analyzer")) {
-            root = root.getParentFile();
-        }
-
-        System.setProperty(Configuration.CLI_WORKING_DIR, root.getCanonicalPath());
 
-        subdir.mkdirs();
-        File settingsGradle = new File(subdir, "settings.gradle");
-        settingsGradle.createNewFile();
-        SettingsFileIO.writeProjectNameIfNeeded(subdir);
+        SettingsFileIO.findGitDir(subdir);
```

Comment:
Do you also want to add a test case to test that the `IOException` is thrown on no `.git` dir?

Reply:
IOException isn't thrown - its ManipulationUncheckedException

Reply:
Yes, ` testFindGitDir`, sorry!